src/components/application_manager/test/commands/hmi/simple_requests_to_hmi_test.cc

```diff
@@ -0,0 +1,107 @@
+ * without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "gtest/gtest.h"
+#include "utils/shared_ptr.h"
+#include "smart_objects/smart_object.h"
+#include "application_manager/smart_object_keys.h"
+#include "application_manager/commands/commands_test.h"
+#include "application_manager/commands/command.h"
+#include "hmi/request_to_hmi.h"
+#include "hmi/vi_get_vehicle_type_request.h"
+#include "hmi/vi_is_ready_request.h"
+#include "hmi/vi_read_did_request.h"
+#include "hmi/vi_subscribe_vehicle_data_request.h"
+
+namespace test {
+namespace components {
+namespace commands_test {
+namespace hmi_commands_test {
+
+using ::testing::_;
+using ::testing::Types;
+using ::testing::NotNull;
+
+using ::utils::SharedPtr;
+
+namespace commands = ::application_manager::commands;
+using commands::MessageSharedPtr;
+
+class RequestToHMITest : public CommandsTest<CommandsTestMocks::kIsNice> {};
+
+TEST_F(RequestToHMITest, BasicMethodsOverloads_SUCCESS) {
+  SharedPtr<commands::RequestToHMI> command(
+      CreateCommand<commands::RequestToHMI>());
+
+  // Current implementation always return `true`
+  EXPECT_TRUE(command->CleanUp());
+  EXPECT_TRUE(command->Init());
+  EXPECT_NO_THROW(command->Run());
+}
+
+TEST_F(RequestToHMITest, SendRequest_SUCCESS) {
+  SharedPtr<commands::RequestToHMI> command(
+      CreateCommand<commands::RequestToHMI>());
+
+  EXPECT_CALL(this->app_mngr_, SendMessageToHMI(NotNull()));
+
+  command->SendRequest();
+}
+
+template <class Command>
+class RequestToHMICommandsTest
+    : public CommandsTest<CommandsTestMocks::kIsNice> {
+ public:
+  typedef Command CommandType;
+};
+
+typedef Types<commands::VIIsReadyRequest,
+              commands::VIGetVehicleTypeRequest,
+              commands::VIReadDIDRequest,
+              commands::VISubscribeVehicleDataRequest> RequestCommandsList;
+
+TYPED_TEST_CASE(RequestToHMICommandsTest, RequestCommandsList);
+
+TYPED_TEST(RequestToHMICommandsTest, Run_SendMessageToHMI_SUCCESS) {
+  typedef typename TestFixture::CommandType CommandType;
+
+  SharedPtr<CommandType> command = this->template CreateCommand<CommandType>();
+
+  EXPECT_CALL(this->app_mngr_, SendMessageToHMI(NotNull()));
```

Comment:
@LevchenkoS I meant `EXPECT_CALL(this->app_mngr_, SendMessageToHMI(NotNull()));`